components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -360,9 +362,15 @@ private OAuth2AccessTokenRespDTO generateNewAccessToken(OAuthTokenReqMessageCont
         AccessTokenDO newTokenBean = createNewTokenBean(tokReqMsgCtx, oAuthAppBean, existingTokenBean, timestamp,
                 validityPeriodInMillis, oauthTokenIssuer);
         setDetailsToMessageContext(tokReqMsgCtx, validityPeriodInMillis, newTokenBean, timestamp);
-        // Persist the access token in database
-        persistAccessTokenInDB(tokReqMsgCtx, existingTokenBean, newTokenBean, timestamp,
-                newTokenBean.getAccessToken());
+        if (markExistingTokenExpired) {
```

Comment:
Shall we add a comment to describe the if condition? Similar one to the following.
`From the below if condition, it will check whether the existing token needs to be expired or not. Based on that the corresponding parameters will be sent to the persistAccessTokenInDB method.`
WDYT?